src/mrpro/algorithms/optimizers/cg.py

```diff
@@ -93,7 +93,7 @@ def cg(
         if tolerance != 0 and (residual_norm_squared < tolerance**2):
             return solution
 
-        if iteration > 0:
+        if residual_norm_squared_previous is not None:  # not first iteration
```

Comment:
mypy cant determine that residual_norm_squared_previous is not None after the first iteration